integration_tests/test_search.py

```diff
@@ -1,25 +1,2078 @@
+        self.search(
+            "Diary of a minecraft zombie",
+            Common(title=re.compile("(diar|minecraft|zombie)"))
+        )
+
+    def test_unowned_title_pie(self):
+        # NOTE: "Pie Town Woman" isn't in the collection, but there's a book called
+        # "Pie Town," which seems like the clear best option for the first result.
+        # This works on ES6.  On ES1, the first result is instead a book entitled
+        # "The Quiche and the Dead," which has "pie" only in the summary.
+        self.search("Pie town woman", FirstMatch(title="Pie Town"))
+
+    def test_unowned_title_divorce(self):
+        self.search(
+            "The truth about children and divorce", [
+                Common(
+                    audience="adult",
+                    first_must_match=False
+                ),
+                AtLeastOne(
+                    title=re.compile("divorce")
+                )
+            ]
+        )
+
+    def test_unowned_title_decluttering(self):
+        # NOTE: this book isn't in the collection, but the top search results should
+        # be reasonably relevant.  This works on ES6 but fails on ES1.
+
+        self.search(
+            "Decluttering at the speed of life", [
+                AtLeastOne(title=re.compile("declutter")),
+                Common(subject=re.compile(
+                    "(house|self-help|self-improvement|decluttering|organization)"
+                ), first_must_match=False)
+            ]
+        )
+
+    def test_unowned_partial_title_rosetta_stone(self):
+        # NOTE: the collection doesn't have any books with titles containing "rosetta"
+        # stone," but it does have a few with titles containing "rosetta"; ideally,
+        # one of those would be the first result.  A title containing "stone" would be
+        # less relevant to the user, but still reasonable.  Instead, the first result
+        # is a memoir by an author whose first name is "Rosetta."
+
+        self.search(
+            "Rosetta stone",
+            FirstMatch(title=re.compile("(rosetta|stone)"))
+        )
+
+    def test_unowned_misspelled_partial_title_cosmetics(self):
+        # NOTE: this fails on both versions of ES.  The user was presumably
+        # looking for "Don't Go to the Cosmetics Counter Without Me," which
+        # isn't in the collection.  Ideally, one of the results should have
+        # something to do with cosmetics; instead, they're about comets.  Fixing
+        # the typo makes the test pass.
+
+        self.search(
+            "Cometics counter", [
+                AtLeastOne(title=re.compile("cosmetics")),
+            ]
+        )
+
+    def test_nonexistent_title_tower(self):
+        # NOTE: there is no book with this title.  The most
+        # likely scenario is that the user meant "The Dark Tower."  This
+        # doesn't currently work on either version of ES.
+
+        self.search("The night tower", FirstMatch(title="The Dark Tower"))
```

Comment:
It seems significantly less important for something like this to work where there's no good result. I'd be careful about including things like this and the next test in the comparison between the versions.

Reply:
I think we can take out the "a hand to reach" test altogether. It's really unclear what the patron is searching for there. 

Here there is a good result, we're just not sophisticated enough to provide it. If you search Google for "the night tower" it shows you _The Dark Tower_ results, and I think that's probably what this person was searching for. This might be past the edge of where we can reasonably expect to get with our search engine, though.